Library/ViewModels/PledgeViewModelTests.swift

```diff
@@ -1223,46 +1273,92 @@ final class PledgeViewModelTests: TestCase {
           |> Backing.lens.amount .~ 700
       )
 
-    self.confirmButtonHidden.assertDidNotEmitValue()
-    self.confirmButtonEnabled.assertDidNotEmitValue()
+    self.submitButtonTitle.assertDidNotEmitValue()
+    self.submitButtonEnabled.assertDidNotEmitValue()
 
     self.vm.inputs.configureWith(project: project, reward: reward, refTag: nil, context: .update)
     self.vm.inputs.viewDidLoad()
 
-    self.confirmButtonHidden.assertValues([false])
-    self.confirmButtonEnabled.assertDidNotEmitValue()
+    self.submitButtonTitle.assertValues(["Confirm"])
+    self.submitButtonEnabled.assertValues([false])
+
+    self.vm.inputs.pledgeAmountViewControllerDidUpdate(
+      with: (amount: 690, min: 25.0, max: 10_000.0, isValid: true)
+    )
+
+    self.vm.inputs.shippingRuleSelected(.init(cost: 1, id: 1, location: .brooklyn))
+
+    self.submitButtonEnabled.assertValues([false, false], "Shipping rule and amount unchanged")
 
     self.vm.inputs.pledgeAmountViewControllerDidUpdate(
       with: (amount: 690, min: 25.0, max: 10_000.0, isValid: true)
     )
 
-    self.confirmButtonHidden.assertValues([false], "Amount unchanged")
-    self.confirmButtonEnabled.assertValues([false], "Amount unchanged")
+    self.submitButtonEnabled.assertValues([false, false, false], "Amount unchanged")
 
     self.vm.inputs.pledgeAmountViewControllerDidUpdate(
       with: (amount: 550, min: 25.0, max: 10_000.0, isValid: true)
     )
 
-    self.confirmButtonHidden.assertValues([false])
-    self.confirmButtonEnabled.assertValues([false, true], "Amount changed")
+    self.submitButtonEnabled.assertValues([false, false, false, true], "Amount changed")
 
     self.vm.inputs.pledgeAmountViewControllerDidUpdate(
       with: (amount: 690, min: 25.0, max: 10_000.0, isValid: true)
     )
 
-    self.confirmButtonHidden.assertValues([false])
-    self.confirmButtonEnabled.assertValues([false, true, false], "Amount unchanged")
+    self.submitButtonEnabled.assertValues([false, false, false, true, false], "Amount unchanged")
 
     self.vm.inputs.shippingRuleSelected(.template)
 
-    self.confirmButtonHidden.assertValues([false])
-    self.confirmButtonEnabled.assertValues([false, true, false, true], "Shipping rule changed")
+    self.submitButtonEnabled.assertValues([false, false, false, true, false, true], "Shipping rule changed")
 
     self.vm.inputs.shippingRuleSelected(.init(cost: 1, id: 1, location: .brooklyn))
 
-    self.confirmButtonHidden.assertValues([false])
-    self.confirmButtonEnabled.assertValues(
-      [false, true, false, true, false], "Amount and shipping rule unchanged"
+    self.submitButtonEnabled.assertValues(
+      [false, false, false, true, false, true, false], "Amount and shipping rule unchanged"
+    )
+  }
+
+  func testUpdatingConfirmButtonEnabled_NoShipping() {
```

Comment:
Note to self that I need to add a test here for `testUpdatingConfirmButtonEnabled_ChangePaymentMethod` 😅